src/main/java/org/checkerframework/checker/dependencyinjection/DependencyInjectionTransfer.java

```diff
@@ -41,8 +48,47 @@ public TransferResult<CFValue, CFStore> visitMethodInvocation(
               boundClassTypeMirror.getAnnotation(), diATF.classValValueElement, String.class);
 
       accumulate(node, result, classNames.toArray(new String[1]));
+    } else if (diATF.isAnnotatedWithMethod(node.getTree())) {
+      MethodInvocationNode methodInvocationNode = node;
+      MethodInvocationNode callToBindNode =
+          (MethodInvocationNode) methodInvocationNode.getTarget().getReceiver();
+      MethodInvocationNode namesClassNode =
+          (MethodInvocationNode) methodInvocationNode.getArgument(0);
+      FieldAccessNode boundClass = (FieldAccessNode) callToBindNode.getArgument(0);
+      StringLiteralNode givenName = (StringLiteralNode) namesClassNode.getArgument(0);
+
+      accumulateBindAnnotatedWith(
+          node, result, boundClass.getReceiver().getType().toString(), givenName.getValue());
     }
 
     return result;
   }
+
+  public void accumulateBindAnnotatedWith(
+      Node node, TransferResult<CFValue, CFStore> result, String value, String name) {
+    List<String> valuesAsList = Arrays.asList(value);
+    List<String> namesAsList = Arrays.asList(name);
+    // If dataflow has already recorded information about the target, fetch it and integrate
+    // it into the list of values in the new annotation.
+    JavaExpression target = JavaExpression.fromNode(node);
+
+    AnnotationMirror newAnno = diATF.createBindAnnotatedWithAnnotation(valuesAsList, namesAsList);
+
+    // TODO: insertValue not updating stores, insertValuePermitNondeterministic is
```

Comment:
We should figure out why that is. I would expect that the `@Pure` annotations on the appropriate methods in the stub file should fix this issue (since that means they are deterministic). Are there other methods being called in the example code that might be non-deterministic and need purity stubs? `Names.named()` comes to mind as a possibility.

Reply:
I didn't think about this but it's definitely a possibility. I'll try this out.

Reply:
Discussed this in a meeting. Possibly something to come back to in the future, but using calls to `insertValuePermitNondeterministic` are okay for now.

[Issue](https://github.com/jaysonacosta/guice-checker/issues/24)